R/visit.R

```diff
@@ -43,3 +43,166 @@ visit_one <- function(pd_flat, funs) {
   reduce(funs, function(x, fun) fun(x),
          .init = pd_flat)
 }
+
+
+#' Propagate context to terminals
+#'
+#' Implements a very specific pre-visiting scheme, namely to propagate
+#'   indention, spaces and lag_newlines to inner token to terminals. This means
+#'   that information regarding indention, linebreaks and spaces (which is
+#'   relative in `pd_nested`) will be converted into absolute.
+#' @inherit context_towards_terminals
+#' @seealso context_towards_terminals visitors
+context_to_terminals <- function(pd_nested,
```

Comment:
I tried that but I could not find an easy way to do that. In contrast to `pre_visit()`, we pass scalars from one level to the other, not functions. Also, unlike in `pre_visit()`, we do not map over the children only, but simultaneously over other columns too `pmap()` instead of `map()`. You could probably create a function that can accommodate both, but I felt it would not make things clearer since the tasks are not very similar in their implementation.